code/game/objects/items.dm

```diff
@@ -1732,3 +1732,17 @@
 	if(!isnull(loc))
 		SEND_SIGNAL(loc, COMSIG_ATOM_CONTENTS_WEIGHT_CLASS_CHANGED, src, old_w_class, new_w_class)
 	return TRUE
+
+/**
+ * Used to determine if an item should be considered contraband by N-spect scanners or scanner gates.
+ * Returns true when an item has the contraband trait, or is included in the traitor uplink.
+ */
+/obj/item/proc/is_contraband()
+	if(HAS_TRAIT(src, TRAIT_CONTRABAND))
+		return TRUE
+	for(var/datum/uplink_item/traitor_item as anything in subtypesof(/datum/uplink_item))
+		if(ispath(src.type, traitor_item.item))
```

Comment:
Also how is this even working? You're accessing `.item` from a typepath, not an instanced object

I think you should be iterating `SStraitor.uplink_items` instead, so you can access instances of uplink items